src/components/movie-form/index.jsx

```diff
@@ -1,93 +1,82 @@
 import './style.css'
 
 import React from 'react'
-import { useParams, useNavigate } from 'react-router-dom'
+import { useParams, useNavigate, useLocation } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
 
 import Input from '../../common/input'
 import Button from '../../common/button'
 
 import { useMoviesContext } from '../../contexts'
-import { formatRuntime } from '../../utils'
-
-const createMovieData = (form, movieId) => {
-    return {
-        id: movieId || window.crypto.randomUUID(),
-        name: form.title.value,
-        releaseDate: form.date.value,
-        image: form.url.value,
-        genres: [form.genre.value],
-        duration: form.time.value,
-        description: form.overview.value,
-        rating: form.rating.value
-    }
-}
+import { addMovie, updateMovie } from '../../utils'
 
 const MovieForm = () => {
     const {id} = useParams()
-    const {genres, moviesData, updateMovie, addMovie} = useMoviesContext()
-    
-	const navigate = useNavigate()
+    const {genres, moviesData} = useMoviesContext()
 
+    const location = useLocation()
+	const navigate = useNavigate()
+    const previousUrl = [location.state?.previousLocation?.pathname, location.state?.previousLocation?.search].join('')
+    const validationRules = { required: true }
+    
     const movies = moviesData?.data
     const details = movies?.find((movie) => movie.id === Number(id))
-    const defaultGenre = genres.find((genre) => details?.genres?.map((genre) => genre.toLowerCase()).includes(genre.name))
-    const runtime = details?.runtime && formatRuntime(details.runtime)
 
-    const resetForm = (event) => {
-        event.preventDefault()
-        event.currentTarget.parentElement.parentElement.reset()
-    }
+    const {register, handleSubmit, reset} = useForm({
+        defaultValues: {
+            title: details?.title,
+            release_date: details?.release_date,
+            poster_path: details?.poster_path,
+            vote_average: details?.vote_average,
+            genres: details?.genres,
+            runtime: details?.runtime,
+            overview: details?.overview
+        }
+    })
 
-    const handleSubmit = (event) => {
-        event.preventDefault()
-    
-        const form = event.currentTarget
-		const dataToSend = createMovieData(form, details?.id)
+    const onSubmit = async (values) => {
+        values.release_date = values.release_date.toISOString().split('T')[0]
 
-        details?.id ? updateMovie(dataToSend) : addMovie(dataToSend)
-        navigate('/')
+        details?.id ? await updateMovie({id: details.id, ...values}) : await addMovie(values)
+        navigate(previousUrl)
+        navigate(0)

```

Comment:
why do you need 2 navigates?

Reply:
I use one to return to a certain url and the second one to refresh the page so that a new call is made to fetch movies data with updated data. I won't to omit use useState to trigger rerender and data fetch